### ISSUE TITLE
Sharing workflow / running workflows as an non-registered user

### ISSUE THREAD

**Issue by joachimwolff** [galaxyproject/galaxy#5898]
I tried to run and or import a workflow as a non-registered user and I get the error `You must be logged in to to import a workflow.`. Also tried to access it via id directly but there I got `API authentication required for this request`. 

Is there a way to use workflows as a non-registerd user that I (and @erasche :) ) do no know so far? If no, this would be a really nice to have feature especially if you want to use the Galaxy platform as a basis for a publication. Many magazines require that users don't have to register.

Thanks for considering.

**Comment by nsoranzo** [comment]
That would only be possible if we allow a user to run a shared workflow without importing it.

**Comment by hexylena** [comment]
I'd certainly be in favour of that. (It would help get closer to my ideal that users shouldn't be duplicating a workflow unless they plan to edit it/need it to be forever unchanged.) Otherwise it gets out of sync with whoever is actively developing the workflow.

**Comment by martenson** [comment]
@joachimjacob anons are not allowed to build/operate on workflows at all, I am in favor of considering a change though. 

However does it makes sense with the limited quota they usually get on public instances?

**Comment by joachimwolff** [comment]
It makes sense because journals require that it is not mandatory to register. I agree that users cannot do much than but this is irrelevant as long as the journal is happy and we can publish.

**Comment by martenson** [comment]
So if they couldn't run it as anons what does the journal need in order to publish? See an image of the workflow builder interface?

**Comment by hexylena** [comment]
It's less the journal needs to publish and more that the reviewers need to use/explore the tool in order to review in order to publish. Seeing an image of the workflow doens't give the same experience as being able to run the tools in a pre-configured workflow where reviewers can't screw it up

**Comment by martenson** [comment]
@erasche Right, so this is about anons being able to execute an actual run of a workflow from manuscript? They will often encounter the quota limit, won't they? 

Could they be given a temporary credentials instead?

**Comment by nsoranzo** [comment]
@martenson Quota limit can be configured (or removed) by the Galaxy admin to allow reviewers to complete a test analysis. Temporary credentials can probably be traced to the reviewer.

**Comment by hexylena** [comment]
exactly as @nsoranzo said on both points. we configure a quota but expect that a reviewer running the predefined workflow on the provided test datasets will not reach that value.

**Comment by martenson** [comment]
Thanks for explanations. Now I think this is decent use case for allowing anons to manipulate with workflows. We can save all the info with their session like anything else.

**Comment by hexylena** [comment]
Just a clarification, I'm not proposing that anons should be able to *manipulate* (I'm reading this word as "edit") workflows, only run existing ones. Maybe it's easier to implement that way though so whatever works. (But I still dream of view+run-only as the default published workflow running mode.)

**Comment by martenson** [comment]
I think we should take this further and remove the registration restriction of workflow creating/editing/importing/running. Afaik workflows are underused and this could help with exposure.
